File: 1684-count-the-number-of-consistent-strings/1684-count-the-number-of-consistent-strings.java
import java.util.regex.Pattern;
class Solution {
    public int countConsistentStrings(String allowed, String[] words) {
        String regex = String.format("^[%s]*$", allowed);
        Pattern pattern = Pattern.compile(regex);
        int count = 0;
        for(String word : words) {
            if (pattern.matcher(word).matches()) count++;
        }
        return count;
    }
}

File: 1684-count-the-number-of-consistent-strings/1684-count-the-number-of-consistent-strings.kt
class Solution {
fun countConsistentStrings(allowed: String, words: Array<String>): Int {
    var count = 0
    for (word in words) {
        var isConsistent = true
        for (c in word) {
            if (!allowed.contains(c)) {
                isConsistent = false
                break
            }
        }
        if (isConsistent) {
            count++
        }
    }
    return count
}
}